### PR TITLE
Resolve #1951 -- Replace Opbeat logging with Sentry

### LINKED ISSUE #1951: Replace Opbeat
Opbeat is closing May 24, 2018. We need to find and add a replacement to track exceptions and performance. 

Potential alternatives:

- [Elastic APM](https://www.elastic.co/solutions/apm)
- [Sentry](https://sentry.io/)

_Please add other potential alternatives below_

### PR DESCRIPTION
**Pull request by oliverroick**
#### Why I made this change

Opbeat is closing down, so we need another way to track errors besides log files. Sentry provides a similar service. 

#### Description of the change

- Remove all references to Opbeat logging. This includes:
	1. Opbeat settings in `production.txt`
	2. Logging Twilio errors with Opbeat. Instead of logging exceptions using the Sentry client, we're logging an exception, which will cause the root logger to trigger an event with Sentry.
- Add Sentry logging to `settings/production.py`
- Add `sentry_dsn` to production env
- Fix up the order of Ansible production playbooks. This was necessary so we don't have to install `raven` on dev machines. `raven` however is imported in `settings/production.py`, which itself is accessed in the `cadasta/application` playbook to run database migrations. To fix that, we need to make sure that the production settings are installed beforehand. 

#### How someone else can test the change

It's not testable in dev environments. The branch is currently installed on `staging`, the only way to test things is to deliberately trigger an exception. 

### When should this PR be merged

https://github.com/Cadasta/cadasta-workertoolbox/pull/7 is related and should be released before this PR is merged,

### Risks

There's a low risk that Sentry is not configured correctly. We need to watch the error logs for some time to ensure everything works as intended. 

### Follow-up actions

- [ ] Remove Opbeat info from environment configs

#### General

**Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 

- [ ] Review 1
- [ ] Review 2

**Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 

- [ ] Review 1
- [ ] Review 2

**Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

- [ ] Review 1
- [ ] Review 2

#### Functionality

**Are all requirements met?** Compare implemented functionality with the requirements specification.

- [ ] Review 1
- [ ] Review 2

**Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

- [ ] Review 1
- [ ] Review 2

#### Code

**Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.

- [ ] Review 1
- [ ] Review 2

**Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 

- [ ] Review 1
- [ ] Review 2

**Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

- [ ] Review 1
- [ ] Review 2

**Is the code documented sufficiently?** Large and complex classes, functions or methods must be annotated with comments following our [code-style guidelines](https://devwiki.corp.cadasta.org/Contributing%20Style%20Guide#documentation-and-comments).

- [ ] Review 1
- [ ] Review 2

**Has the scalability of this change been evaluated?**

- [ ] Review 1
- [ ] Review 2

**Is there a maintenance plan in place?**

- [ ] Review 1
- [ ] Review 2

#### Tests

**Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 

- [ ] Review 1
- [ ] Review 2

**If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.

- [ ] Review 1
- [ ] Review 2

**If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

- [ ] Review 1
- [ ] Review 2

#### Security

**Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**

- [ ] Review 1
- [ ] Review 2

**Are all UI and API inputs run through forms or serializers?** 

- [ ] Review 1
- [ ] Review 2

**Are all external inputs validated and sanitized appropriately?**

- [ ] Review 1
- [ ] Review 2

**Does all branching logic have a default case?**

- [ ] Review 1
- [ ] Review 2

**Does this solution handle outliers and edge cases gracefully?**

- [ ] Review 1
- [ ] Review 2

**Are all external communications secured and restricted to SSL?**

- [ ] Review 1
- [ ] Review 2

#### Documentation

**Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).

- [ ] Review 1
- [ ] Review 2

**Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).

- [ ] Review 1
- [ ] Review 2

**Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki. 

- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

File: cadasta/config/settings/production.py [hunk 1]
```diff
@@ -138,19 +139,23 @@
...
+            'handlers': ['file', 'error_file', 'email_admins'],
             'level': 'DEBUG',
             'propagate': True,
         },
         'xform.submissions': {
-            'handlers': ['file', 'error_file', 'email_admins', 'opbeat'],
+            'handlers': ['file', 'error_file', 'email_admins'],
             'level': 'DEBUG',
             'propagate': True,
         },
```
Comment (alukach) [inline cadasta/config/settings/production.py]: Let's take a moment to double-check that this is actually what we want moving forward.  Is there any reason that we want to treat `xform.submissions` and `core` logs differently than all of our other (and third-party) modules?

If not, I would recommend we get rid of these two loggers and put these handlers onto the `root` logger...

I say this with a shaken level of confidence after my last logging refactor went so poorly, so please push back if this feels wrong.

**Comment by alukach** [comment]
My last change to this PR will cause builds to fail until https://github.com/Cadasta/cadasta-workertoolbox/pull/7 is merged and `0.5.0` is released.

File: cadasta/config/settings/production.py (hunk 1 above)
Comment (oliverroick) [reply cadasta/config/settings/production.py]: You're correct. I changed the code according to your comments and also changed the logging config on staging; works as expected from what I can tell.

**Comment by alukach** [comment]
Released `cadasta-workertoolbox@0.5.0`, tests restarted...

**Pull request merged**